ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -53,23 +92,132 @@ export class ImportedDocuments extends React.Component {
-          data: response.data,
+          data: responseData,
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: responseData,
         });
       }
+      this.modeLabel();
+      this.statusLabel();
+      this.calculateStatistics();
     } else {
       this.intervalId && clearInterval(this.intervalId);
     }
   };
 
+  modeLabel = () => {
+    let { data, mode } = this.state;
+    switch (data.mode) {
+      case 'IMPORT':
+        mode = (
+          <Label color="blue" basic>
+            Import
+          </Label>
+        );
+        break;
+      case 'DELETE':
+        mode = (
+          <Label color="red" basic>
+            Delete
+          </Label>
+        );
+        break;
+      case 'PREVIEW_IMPORT':
+        mode = (
+          <Label color="teal" basic>
+            Preview (import)
+          </Label>
+        );
+        break;
+      case 'PREVIEW_DELETE':
+        mode = (
+          <Label color="teal" basic>
+            Preview (delete)
+          </Label>
+        );
+        break;
+      default:
+        mode = (
+          <Label color="black" basic>
+            Invalid mode
+          </Label>
+        );
+        break;
+    }
+    this.setState({
+      mode: mode,
+    });
+  };
+
+  statusLabel = () => {
+    let { data, status, isLoading } = this.state;
+    !data
+      ? (status = <Label color="grey">Fetching status</Label>)
+      : isLoading
+      ? (status = (
+          <Label color="blue">
+            <Icon loading name="circle notch" /> Importing literature
+          </Label>
+        ))
+      : data.status === 'SUCCEEDED'
+      ? (status = <Label color="green">Import successfull</Label>)
+      : data.status === 'CANCELLED'
+      ? (status = <Label color="yellow">Import cancelled</Label>)
+      : (status = <Label color="red">Import failed</Label>);
+    this.setState({
+      status: status,
+    });
+  };
+
+  calculateStatistics = () => {
+    let { statistics, data, selectedResult } = this.state;
+
+    const dataClone = Array.from(data.records);
+    const newRecords = dataClone.slice(this.lastIndex, dataClone.length);
+
+    this.lastIndex += newRecords.length;
+    statistics.records.value = data.loaded_entries + '/' + data.entries_count;
+
+    let tempRecords = [];
+
+    for (const statistic in statistics) {
+      tempRecords = newRecords.filter(record =>
+        statistics[statistic].filterFunction(record)
+      );
+      statistics[statistic].filteredRecords = statistics[
+        statistic
+      ].filteredRecords.concat(tempRecords);
+      if (statistic !== 'records') {
+        statistics[statistic].value =
+          statistics[statistic].filteredRecords.length;
+      }
+    }
+    this.setState({
+      statistics: statistics,
+    });
+    this.currentFilterResults(selectedResult);
+  };
+
+  currentFilterResults(key) {
+    const { statistics } = this.state;
+    this.setState({
```

Comment:
I am wondering if it would be better to calculate this in the `render` method instead of having a part of the state with all the records, and another part with the filtered. If we have e.g. 1K records, you might have 2 times these 1K records in the state.
We can probably chat about this together.

Reply:
I think if we don't do the filtering on the backend, maybe its better to keep the 2k in the state instead of filter on each render (which in the end will be either way 2k kept in the memory, just different scopes), also for the reasons written in https://github.com/CERNDocumentServer/cds-ils/pull/603#discussion_r719395116